packages/protocol/specs/reserve.spec

```diff
@@ -0,0 +1,27 @@
+pragma specify 0.1
+
+methods {
+  isSpender(address) returns bool envfree
+  isExchangeSpender(address) returns bool envfree
+  getExchangeAddress() returns address envfree
+  contractBalance() returns uint256 envfree
+}
+
+rule nonspenderCantReduceBalance(address a, method f) {
+  require(!isSpender(a));
+  require(!isExchangeSpender(a));
+  require(getExchangeAddress() != a);
```

Comment:
are all of these requirements necessary? Do we know it's safe to assume? In particular the 3rd one

Reply:
Yes, spenders from the `isSpender` mapping and core stable token Exchanges are the exceptional addresses that can spend Reserve's balance. Thanks for pointing this out though, the last two can be simplified into a single call to `isAllowedToSpendExchange`.

Reply:
Ah, actually `isAllowedToSpendExchange` is a modifier so the code can't be simplified that way.